File: incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/extension/MemberRole.kt
package me.proton.drive.sdk.extension

import me.proton.drive.sdk.entity.MemberRole
import proton.drive.sdk.ProtonDriveSdk

fun ProtonDriveSdk.MemberRole.toEntity() = when (this) {
    ProtonDriveSdk.MemberRole.MEMBER_ROLE_INHERITED -> MemberRole.INHERITED
    ProtonDriveSdk.MemberRole.MEMBER_ROLE_VIEWER -> MemberRole.VIEWER
    ProtonDriveSdk.MemberRole.MEMBER_ROLE_EDITOR -> MemberRole.EDITOR
    ProtonDriveSdk.MemberRole.MEMBER_ROLE_ADMIN -> MemberRole.ADMIN
    ProtonDriveSdk.MemberRole.UNRECOGNIZED -> error("Invalid MemberRole: $this")
}

File: client/cs/src/Proton.Drive.Sdk/Nodes/MemberRole.cs
namespace Proton.Drive.Sdk.Nodes;

/// <summary>
/// Role of the user on a node, granting a set of permissions.
/// </summary>
public enum MemberRole
{
    /// <summary>
    /// The role is inherited from an ancestor node; the node is not shared directly with the user.
    /// </summary>
    Inherited = 0,

    /// <summary>
    /// The user can view the node but not modify it.
    /// </summary>
    Viewer = 1,

    /// <summary>
    /// The user can view and modify the node.
    /// </summary>
    Editor = 2,

    /// <summary>
    /// The user can view, modify and manage sharing of the node.
    /// </summary>
    Admin = 3,
}
